src/backend/distributed/master/master_metadata_utility.c

```diff
@@ -1277,37 +1277,36 @@ UpdateColocationGroupReplicationFactor(uint32 colocationId, int replicationFacto
 	/* we first search for colocation group by its colocation id */
 	pgDistColocation = heap_open(DistColocationRelationId(), RowExclusiveLock);
 	tupleDescriptor = RelationGetDescr(pgDistColocation);
-	ScanKeyInit(&scanKey[0], Anum_pg_dist_colocation_colocationid, BTEqualStrategyNumber,
-				F_OIDEQ, ObjectIdGetDatum(colocationId));
+	ScanKeyInit(&scanKey[0], Anum_pg_dist_colocation_distributioncolumntype,
+				BTEqualStrategyNumber,
+				F_OIDEQ, ObjectIdGetDatum(InvalidOid));
 
 	scanDescriptor = systable_beginscan(pgDistColocation,
-										DistColocationColocationidIndexId(), indexOK,
+										InvalidOid, indexOK,
```

Comment:
Why have we changed `DistColocationColocationidIndexId ()` to  `InvalidOid ` ? That's the Oid of the index relation that Postgres uses for the systable scan. Is that intentional? If so, we should consider making `indexOK = false` as well. But, I think we should just add back `DistColocationColocationidIndexId ()`.

Reply:
Because I changed the search to be based on a column which is not indexed

Reply:
Ok, makes sense. Let's set `indexOK  = false` in that case.